lib/pavilion/output.py

```diff
@@ -71,25 +71,32 @@
 }
 
 
-def get_relative_timestamp(base_time: float) -> str:
+def get_relative_timestamp(base_time, fullstamp=False):
     """Print formatted time string based on the delta of time objects.
-    :param base_time: The unix timestamp
+    :param float base_time: The datetime object to compare and format
+    from.
+    :param bool fullstamp: Whether to return the whole formatted
+    timestamp.
     :returns: A formatted time string.
     """
 
     if not isinstance(base_time, float):
-        return ''
+        return base_time
 
-    now = datetime.datetime.now()
-    format_ = ['%Y', '%b', '%a', '%H:%M:%S']  # year, month, day, time
+    #Duration
+    if base_time < 1e5:
+        return str(datetime.timedelta(seconds=round(base_time)))
 
+    now = datetime.datetime.now()
     base_time = datetime.datetime.fromtimestamp(base_time)
-
-    for i in range(0, len(format_)):
-        if now.strftime(format_[i]) != base_time.strftime(format_[i]):
-            return base_time.strftime(" ".join(format_[i:]))
-
-    return base_time.strftime(str(format_[3]))
+    dateformat = '%Y %b %-d'  # year, month, day
+    timeformat = '%H:%M:%S'  # time
+    final_form = " ".join([dateformat, timeformat])
+    if not fullstamp:
+        if now.date() == base_time.date():
+            final_form = timeformat
+
+    return base_time.strftime(final_form)
```

Comment:
There's a couple reasons.  First off, the original loop was clumsy and unpythonic. Repeated calls to strftime are wasteful. It handles timestamps but doesn't handle duration stamps and '12:13' is much more readable than 733.123312121551454151. The %a format is kind of useless, if you're looking at a test from January in March it's just going to tell you it was some Tuesday back then (unless you've fixed that as it looks like you have from your `pav series ls all` output in the chat).

If I reset output.py to master and look at it on grizzly in my personal pavilion:
``` bash
➜  pavilion2 git:(rslt_keys) ✗ pav status s10
 Test statuses
------+---------------------------------------+----------------------------------------+-------+----------+--------+------------------
 Test | Job id                                | Name                                   | Nodes | State    | Result | Time
------+---------------------------------------+----------------------------------------+-------+----------+--------+------------------
 9    | job_23637_c4d4b46a94_gr-rfe1.lanl.gov | hello_mpi.toss-mpirun.pgi-openmpi      | 1     | COMPLETE | PASS   | Jan Tue 13:10:52
 8    | job_23626_dec53c72b6_gr-rfe1.lanl.gov | hello_mpi.toss-mpirun.intel-openmpi    | 1     | COMPLETE | PASS   | Jan Tue 13:10:52
 7    | job_23612_abef769ac9_gr-rfe1.lanl.gov | hello_mpi.toss-mpirun.gcc-openmpi      | 1     | COMPLETE | PASS   | Jan Tue 13:10:52
 6    | job_23606_ce78c7d5fb_gr-rfe1.lanl.gov | hello_mpi.toss-openmpi.pgi-openmpi     | 1     | COMPLETE | PASS   | Jan Tue 13:10:52
 5    | job_23590_2712ef6d49_gr-rfe1.lanl.gov | hello_mpi.toss-openmpi.intel-openmpi   | 1     | COMPLETE | PASS   | Jan Tue 13:10:52
 4    | job_23587_7212f0e81f_gr-rfe1.lanl.gov | hello_mpi.toss-openmpi.gcc-openmpi     | 1     | COMPLETE | PASS   | Jan Tue 13:10:52
 3    | job_23573_a7f272e4ea_gr-rfe1.lanl.gov | hello_mpi.toss-mvapich2.intel-mvapich2 | 1     | COMPLETE | PASS   | Jan Tue 13:10:52
 2    | job_23569_cb3c3ec70a_gr-rfe1.lanl.gov | hello_mpi.toss-mvapich2.gcc-mvapich2   | 1     | COMPLETE | PASS   | Jan Tue 13:10:52
 1    | job_23558_223130f26e_gr-rfe1.lanl.gov | hello_mpi.intel-mpi.intel-intel-mpi    | 1     | COMPLETE | FAIL   | Jan Tue 13:10:52
```

So I changed it to `%-d`.  And if I was going to change the day I pretty much had to keep the month because 
`11 13:10:52` looks dumb.  And if I was going to keep the day and month regardless why not the year too, because why complicate things. it's only 5 characters (could be 3 because it'd probably be better as a 2 digit year).  So I thought it seemed like a reasonable proposal to just divide things between today and other days.  The original code only split it 4 ways, and I can see a reason to split it 3 ways
* Not this year: '%Y %b %-d %H:%M:%S'
* This year but not today: "%b %-d %H:%M:%S'
* Today: '%H:%M:%S'
but the complexity of the first split to avoid 5 or 3 characters doesn't seem justified.

I added the fullstamp variable to put the complete date on jsons to be saved (but then I split that part off so it was kind of chicken and egg).  Because if you're saving a file, you probably want that for a bit longer term storage and you probably want to look at it without firing up pavilion, to be consumed in your own code.  It'd suck to go back to a file in a year or cite it in a conference paper (because that's where I'm going with this), and find that it was made at 11:55 or 12:13 some Tuesday in some January.  I don't think it neuters the function because there's the default value False so the interface doesn't change for the vast majority of use cases.

Also if you call it with the already transformed timestamp it just does what you wanted it to, which is give you a human readable timestamp. Then you don't have to worry about whether or not the quantity has already been transformed.

That's my justification, but I'm really just throwing it out there. There might be a more elegant way to do it that's more in keeping with your philosophy.

Reply:
I kept the fullstamp variable and used it in results.py.  Otherwise I changed it back to match master.  Ok except for moving some strings to variables.

I moved the timedelta to a separate funciton.